source/common/http/header_utility.cc

```diff
@@ -259,6 +260,22 @@ HeaderUtility::requestHeadersValid(const RequestHeaderMap& headers) {
   return absl::nullopt;
 }
 
+absl::optional<std::pair<std::reference_wrapper<const absl::string_view>, Http::Code>>
+HeaderUtility::requestHeadersValidAtHcm(const RequestHeaderMap& headers,
+                                        uint32_t xff_num_trusted_hops, bool connection_is_ssl) {
+  if (xff_num_trusted_hops == 0 &&
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.add_and_validate_scheme_header") &&
+      !connection_is_ssl &&
```

Comment:
perf nit, I would put `!connection_is_ssl` before the runtime lookup to short circuit faster and avoid the runtime lookup when not needed.